src/mono/wasm/runtime/js_support.js

```diff
@@ -0,0 +1,57 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+// Loads the config file located in the root of the project
+// Not meant to be used outside of this class (TODO make private to this file when project converted to TS)
+function load_config() {
+    // since this file loads before emsdk we don't have environment vars yet, so we define them locally
+    const ENVIRONMENT_IS_NODE = typeof process === "object";
+    const ENVIRONMENT_IS_WEB = typeof window === "object";
+
+    // In some cases there may be no Module object (such as in tests)
+    // so we no-op during the callback
+    const callback = typeof Module !== "undefined" ? Module['onConfigLoaded'] : (_) => {};
+
+    if (ENVIRONMENT_IS_NODE){
+        try {
+            const config = JSON.parse(require("./mono-config.json"));
+            callback(config);
+        } catch(e) {
+            callback({error: "Error loading mono-config.json file from current directory"});
+        }
+    } else if (ENVIRONMENT_IS_WEB){
+        const xobj = new XMLHttpRequest();
```

Comment:
It may be worthwhile to try and use fetch instead. We use it elsewhere and it would be an easier point for user customization

Reply:
This file is added to `dotnet.js` via the emsdk -prejs. Doesn't that mean that fetch is not setup yet? If yes, is there a way to get around this?

Reply:
The consumer would configure fetch before loading dotnet.js. fetch is part of the stdlib

Reply:
Oh so then I misunderstood. This is great! Thanks for pointing it out, I will push an update soon.